### ISSUE TITLE
Memory leak caused by glbinding logging.

### ISSUE THREAD

**Issue by automaticp** [automaticp/ogl-learning#24]
Currently when logging is enabled with `enable_glbinding_logger()` the application leaks memory with speed proportional to number of gl function calls (judging by how it scales with the framerate). Global error checking is suspect, but from looking at my code, I see no reason for a leak. Will have to investigate further.

**Comment by automaticp** [comment]
Looking through the glbinding source code here's what I found:

```C++
void Binding::log(FunctionCall && call)
{
    if (s_logCallback())
    {
        s_logCallback()(new FunctionCall(std::move(call)));
    }
}
```

The allocation itself is largely unnecessary, as it could be just moved further into the actual callback, but it seems like this piece of machinery was not written with the intention to be used by someone else at all, judging by how complex their logging procedure gets with atomic ring-buffers and whatnot. Well, and of course, _it leaks_, if you don't delete the pointer yourself.

Regardless, allocating on every gl call is unlikely the price I can pay in real-time applications, so logging callbacks are a no-no for now. 

Could be be worth replacing the callback signature from `void(FunctionCall* call)` to `void(FucntionCall&& call)` and shifting the responsibility for allocating to the user. Might raise an issue later, idk.

**Comment by automaticp** [comment]
Addressed this in 1e178d5c1bfee76e73584d177779b981ba3a795f, just gonna use global after callback for now.
